@vates/diff/index.js

```diff
@@ -7,7 +7,7 @@
  * @param {Array|Buffer|string} data1 - The first data array or buffer to compare.
  * @param {T} data2 - The second data array or buffer to compare.
  * @param {(index: number, diff: T) => void} [cb] - The callback function to invoke for each difference. If not provided, an array of differences will be returned.
- * @returns {Array<number|T>|undefined} - An array of differences if no callback is provided, otherwise undefined.
+ * @returns {Array<number|T>|undefined} - An array of differences if no callback is provided; otherwise, undefined.
```

Comment:
This isn't technically spelling -- it's grammar, but as I've extended my supplemental rules, I've added checks for things like this.